### ISSUE TITLE
backend kept restart and missing SECRET_KEY

### ISSUE THREAD

**Issue by Chengregnarok** [LibrePhotos/librephotos-docker#60]
Hi there this is a newbee trying to run LibrePhoto on Synology Nas DS720+

I was following all the steps in documentation: https://docs.librephotos.com/1/standard_install/

And when I try to do my 1st login the backend container collapsed...

by reading the error message in container log, it seems like missing SECRET_KEY, screenshot as below

![image](https://user-images.githubusercontent.com/85175548/180646392-4c64a02b-58e3-43a9-ab58-b8b051bb5f15.png)

and by checking the variables of the container, it looks like I'm missing the secret key.. or maybe admin settings as well?

![image](https://user-images.githubusercontent.com/85175548/180646466-47d8e966-fee1-4271-b535-46371c436c8b.png)

hopefully can find some solution for that, or would be fixed by re-install?

**Comment by derneuere** [comment]
You have to use the librephotos.env, edit the values and rename the file to only .env. It should be in the same folder as the docker-compose.yml

The issue is, that the data variable isn't set. If no secret key is found, it will generate one and then will put it in the logs folder. But the logs folder can't be created, if the data variable in the .env isn't set.

If it is not possible with Synology to use the .env then you have to set the ${data} and ${scanDirectory} by hand in the docker-compose.yml

**Comment by Chengregnarok** [comment]
thanks for the comment~

complete following the instruction, but still see below message when running docker compose
![image](https://user-images.githubusercontent.com/85175548/180647887-c9fb3aaa-2356-4d82-8f8f-35df2211c2e3.png)

is that right?

and I noticed below can only be set in docker-compose.yml
![image](https://user-images.githubusercontent.com/85175548/180648021-365d9b68-075e-403c-acd2-f2dacb489409.png)

not sure if I need to make a change...

cuz I don't have a secret_key file under logs folder...

**Comment by derneuere** [comment]
Yes, that is okay. These are old values, which are no longer needed. The secret key gets created when it's starting up.

**Comment by Chengregnarok** [comment]
retried, but seems not fixing the problem.
[backend.zip](https://github.com/LibrePhotos/librephotos-docker/files/9175967/backend.zip)
would appreciate if can help read my backend log to see where the problem is... still working on my windows WSL to check if that is the problem of synology...

**Comment by Chengregnarok** [comment]
facing problem when working on WSL..

![image](https://user-images.githubusercontent.com/85175548/180653053-edf3ac71-e779-4b8c-9732-580f42345abc.png)

**Comment by derneuere** [comment]
Can you show me the folder, where the docker-compose.yml is?

**Comment by Chengregnarok** [comment]
the synology nas one: /root/librephotos-docker
![image](https://user-images.githubusercontent.com/85175548/180654482-ef321b04-7d0b-4037-b8c0-7bbd75a622ab.png)

the Windows WSL one: /root/librephotos-docker
![image](https://user-images.githubusercontent.com/85175548/180654534-73dfe90f-9b55-4f2c-938e-fc6b3c1ab894.png)

**Comment by derneuere** [comment]
Alright, this looks correct. I think the real error is happening on boot up. I would guess that the permission of the data folder is wrong. Can you get me the log direct after boot up of the backend?

**Comment by Chengregnarok** [comment]
I think for Synology I've granted full rights to all users.
Synology didn't see any error message after run docker-compose up -d. But...
before I login in port 3000, the backend container collapse, below the html file I downloaded from synology docker directly after collapse.
[backend (1).zip](https://github.com/LibrePhotos/librephotos-docker/files/9176082/backend.1.zip)

for WSL, I didn't even make the backend container created. Below the error message
![image](https://user-images.githubusercontent.com/85175548/180655003-ff2652bc-ca02-44f4-bc13-7d0daad5ca4a.png)
and this time it is db container not working seems... below snapshot from docker windows version

![image](https://user-images.githubusercontent.com/85175548/180655037-39b93e7d-d9eb-4ed8-86d9-35a4c95ec554.png)
